library/src/main/scala/scala/scalajs/runtime/LinkingInfo.scala

```diff
@@ -16,7 +16,23 @@ import scala.scalajs.js
 
 /** Information about link-time configuration of Scala.js. */
 sealed trait LinkingInfo extends js.Object {
-  /** Whether we are assuming ECMAScript 6 support or not. */
+  /** Version (edition) of ECMAScript that is assumed to be supported by the
+   *  runtime.
+   *
+   *  This is an integer that represents the *edition* of the ECMAScript
+   *  Language Specification. For example, ECMAScript 2015 is represented with
+   *  the value `6`.
+   *
+   *  As an exception, ECMAScript 5.1 is represented with the value `5`.
+   */
+  val esVersion: Int
+
+  /** Whether Scala.js language features use ECMAScript 2015 (edition 6)
+   *  semantics or not.
+   *
+   *  For historical reasons, this is called `assumingES6`, but a better name
+   *  would be `useECMAScript2015Semantics`.
```

Comment:
Just to check my understanding: We cannot re-name this, otherwise a new linker will not properly link an old library anymore, right?

Reply:
Yes, that's right.